### PR TITLE
HARMONY-1414: Added support to harmony requests that completes successfully without creating a job.

### PR DESCRIPTION
**Pull request by ygliuvt**
## Jira Issue ID
HARMONY-1414

## Description
Added support to harmony requests that completes successfully without creating a job.

## Local Test Steps
Run last demo of tutorial notebook and verify the result returned from harmony client submission call is the download links in the harmony request response.

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [ ] Documentation updated (if needed)

### CODE REVIEW

File: harmony/harmony.py [hunk 1]
```diff
@@ -802,8 +802,10 @@ def submit(self, request: BaseRequest) -> any:
         if response.ok:
             if isinstance(request, CapabilitiesRequest):
                 return response.json()
+            elif response.json()['status'] == 'successful':
+                return response.json()['links']
```
Comment (chris-durbin) [inline harmony/harmony.py:806]: We should return the full JSON response for the no op service. When there's a job the user can use the job ID to pull back the full job status information, but in this case there's no way to get the other information with another call. I'd want to see the message for example that the download links do not reflect the subsetting I requested.

**Review by chris-durbin — CHANGES_REQUESTED** [review]
I want to make sure a user can use the response similar to if a job were returned. In order to do that, I think we should update the response to include the full JSON rather than just the links. We should also update the download functions (including download_all) to take that JSON response and download the links. Ideally we can also make the wait_for_processing (or similarly named) function also work with that workflow.

**Comment by ygliuvt** [comment]
> I want to make sure a user can use the response similar to if a job were returned. In order to do that, I think we should update the response to include the full JSON rather than just the links. We should also update the download functions (including download_all) to take that JSON response and download the links. Ideally we can also make the wait_for_processing (or similarly named) function also work with that workflow.

These are valid points and also completely changed the scope of the ticket. I will see how the download function works for directly download, update the ticket and implementation. Thanks.

**Review by indiejames — APPROVED** [review]

**Review by chris-durbin — APPROVED** [review]
Tested out successfully.

**Pull request merged**